File: src/main/java/com/github/aivancioglo/resttest/setters/Body.java
package com.github.aivancioglo.resttest.setters;

import com.github.aivancioglo.resttest.http.HTTPRequest;

public class Body<T extends HTTPRequest> implements Setter {
    private Object body;

    public Body(Object body) {
        this.body = body;
    }

    /**
     * Update request.
     *
     * @param request actual request.
     */
    @Override
    public void update(HTTPRequest request) {
        request.setBody(body);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/setters/Body.kt
package com.github.aivancioglo.resttest.setters

import com.github.aivancioglo.resttest.http.HTTPRequest

/**
 * Request body setter.
 *
 * @constructor is creating variable with requestSpecification body value.
 * @param body of your requestSpecification.
 */
class Body<in T : HTTPRequest<*>>(private val body: Any) : Setter<T> {

    /**
     * Setting body of your requestSpecification.
     *
     * @param request that will be updated.
     */
    override fun update(request: T) {
        request.requestSpecification.body(body)
    }
}